src/proxy.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { URL } from 'url';
+import { ScannerProperties, ScannerProperty } from './types';
+
+export function getProxyUrl(properties: ScannerProperties): URL | undefined {
+  const proxyHost = properties[ScannerProperty.SonarScannerProxyHost];
+  const serverUsesHttps = properties[ScannerProperty.SonarHostUrl].startsWith('https');
```

Comment:
I see that we will also need to support SC for proxy
https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-net/

I've been thinking we need a property or way of knowing the TRUE host URL through the whole process, without checking every time

for example, if they set the SonarcloudUrl property, perhaps at start we should use that to set this SonarHostUrl so we can rely on it

Reply:
>  if they set the SonarcloudUrl property, perhaps at start we should use that to set this SonarHostUrl so we can rely on it

I think it's a good idea, will change this PR to support this